### DISCUSSION TITLE
Bad request

### DISCUSSION THREAD

**Discussion by gbralo** [mivano/azure-cost-cli#3] [General]
i am getting this error
Error: Response status code does not indicate success: 400 (Bad Request).
i specified subscription id

**Comment by mivano** [comment]
Hmm that does not provide a lot of details. I will make sure we get some additional logging. Maybe that helps. You see this on a specific/single subscription id or on all?

**Comment by mivano** [comment]
@gbralo the 0.6.0 version contains a `--debug` option. If you would like to retry and pass in this option, it can show the request and response. That may help to find out why you get the bad request.

**Comment by NewtAdmin2** [comment]
Getting bad request 400 :

Response status code is BadRequest and got payload size of 905
Response content: {"error":{"code":"BadRequest","message":"Invalid query definition: Invalid dataset grouping: 'BillingPeriod'; valid values:
'ResourceGroup','ResourceGroupName','ResourceType','ResourceId','ResourceLocation','SubscriptionId','SubscriptionName','MeterCategory','MeterSubcategory','M
eter','ServiceFamily','UnitOfMeasure','PartNumber','BillingAccountName','BillingProfileId','BillingProfileName','InvoiceSection','InvoiceSectionId','Invoice
SectionName','Product','ResourceGuid','ChargeType','ServiceName','ProductOrderId','ProductOrderName','PublisherType','ReservationId','ReservationName','Freq
uency','InvoiceId','PricingModel','CostAllocationRuleName','MarkupRuleName','BillingMonth','Provider','BenefitId','BenefitName','CustomerTenantId','Customer
TenantDomainName','ResellerMPNId','PartnerEarnedCreditApplied','CustomerName','PartnerName',''.\r\n\r\n (Request ID:
25fe2287-859d-4741-a62a-8856f5018de1)"}}

**Comment by mivano** [comment]
That looks like this issue: https://github.com/mivano/azure-cost-cli/issues/6 so lets take it from there.

**Reply by gbralo** [reply]
update to 0.9.0 does the job, thanks!

**Reply by mivano** [reply]
Great!, thanks for the feedback.

**Comment by NewtAdmin2** [comment]
Hi, Thanks for sharing the solution. After upgrading to 0.9.0, solution works.

But, I am not able to get the information such as By Service Name, Location, Resource Group.
![image](https://user-images.githubusercontent.com/46739507/232697894-4971a050-4a55-4f30-8c3b-9fd37536ad39.png)

Tried in both json and text also..

**Reply by mivano** [reply]
Looks similar to this issue: https://github.com/mivano/azure-cost-cli/issues/16
Can you confirm your quota id?
